lib/context-base.js

```diff
@@ -5,18 +5,28 @@
 
 'use strict';
 
+var assert = require('assert');
 var EventEmitter = require('events').EventEmitter;
+var TypeRegistry = require('./type-registry');
 var inherits = require('util').inherits;
 
 module.exports = ContextBase;
 
 /**
  * A base class for all Context instances
  */
-function ContextBase(method) {
+function ContextBase(method, typeRegistry) {
+  // NOTE(bajtos) we are not asserting via "instanceof" to allow
+  // multiple copies of strong-remoting to cooperate together
+  assert(method && typeof method === 'object',
+    'method must be a SharedClass instance');
+  assert(typeRegistry && typeof typeRegistry === 'object',
+    'typeRegistry must be a TypeRegistry instance');
```

Comment:
We have used them a lot in `dao.js` as well; this `assert(...)` things are so neat; however I just wonder if the assertion fails, does it dump the stack trace?